lib/tools.js

```diff
@@ -2410,17 +2410,25 @@ function parseGithubPathname(pathname) {
 
 /**
  * Removes properties which are given by preserve
- * @param {object} preserve - object which has true entries for all attributes which should be removed from currObj
+ * @param {object} preserve - object which has true entries (or array of selected attributes) for all attributes which should be removed from currObj
  * @param {object} oldObj - old object
  * @param {object} newObj - new object
  */
 function removePreservedProperties(preserve, oldObj, newObj) {
     for (const prop of Object.keys(preserve)) {
         if (isObject(preserve[prop]) && newObj[prop]) {
+            // we have to go one step deeper
             removePreservedProperties(preserve[prop], oldObj[prop], newObj[prop]);
         } else if (newObj && newObj[prop] !== undefined && (!oldObj || oldObj[prop] !== undefined)) {
             // newObj has the prop and oldObj has it -> remove from newObj to preserve
```

Comment:
The comment does not match the logic above. If `oldObj` is undefined and `newObj` has the property, it will be removed. Is that how it is meant to function?

Reply:
True, mixed it with extendObject logic not needed here